src/main/java/org/junit/runners/model/MultipleFailureException.java

```diff
@@ -33,9 +33,13 @@ public MultipleFailureException(List<Throwable> errors) {
         this.fErrors = new ArrayList<Throwable>(errors.size());
         for (Throwable error : errors) {
             if (error instanceof AssumptionViolatedException) {
-                error = new TestCouldNotBeSkippedException((AssumptionViolatedException) error);
+                fErrors.add(
+                        new TestCouldNotBeSkippedException((AssumptionViolatedException) error));
+            } else if (error instanceof MultipleFailureException) {
```

Comment:
Any thrown `MultipleFailureException` will be unwrapped, recursively, in `EachTestNotifier.addFailure()`. Are there any benefits to doing it here as well?